lib/appium/AppiumRunner.js

```diff
@@ -0,0 +1,483 @@
+
+    if (rule) {
+        logger.normal("paramedic-appium: It is already in place");
+    } else {
+        rule = new elementTree.Element("access");
+        rule.attrib.origin = origin;
+        xml.getroot().append(rule);
+        fs.writeFileSync(configFile, xml.write({indent: 4}), util.DEFAULT_ENCODING);
+    }
+}
+
+function runCommand(command, appPath) {
+    if (appPath) {
+        shell.pushd(appPath);
+    }
+    shell.exec(command);
+    if (appPath) {
+        shell.popd();
+    }
+}
+
+function isFailFastError(error) {
+    if (error && error.message) {
+        return error.message.indexOf("Could not find a connected") > -1 ||
+            error.message.indexOf("Bad app") > -1;
+    }
+    return false;
+}
+
+function killProcess(procObj, killSignal, callback) {
+    if (procObj && procObj.alive) {
+        procObj.alive = false;
+        setTimeout(function () {
+            kill(procObj.process.pid, killSignal, callback);
+        }, 1000);
+    } else {
+        callback();
+    }
+}
+
+function installAppiumServer(success, fail) {
+    logger.normal("paramedic-appium: Installing Appium server...");
+    shell.pushd(path.join(__dirname, "../.."));
+    shell.exec("npm install appium", { async: true, silent: !global.VERBOSE }, function (code, stdout) {
+        if (code !== 0) {
+            fail("Couldn't install Appium server:\n" + stdout);
+        } else {
+            success();
+        }
+    });
+    shell.popd();
+}
+
+AppiumRunner.prototype.cleanUp = function (callback) {
+    var self = this;
+
+    killProcess(self.appium, KILL_SIGNAL, function () {
+        killProcess(self.iosProxy, KILL_SIGNAL, function () {
+            callback();
+        });
+    });
+};
+
+AppiumRunner.prototype.startTests = function (callback) {
+    var jasmine = new Jasmine();
+    var self = this;
+
+    function exitGracefully(e) {
+        if (!!e) {
+            logger.normal("paramedic-appium: " + e);
+        }
+        logger.normal("paramedic-appium: Uncaught exception! Killing server and exiting in 2 seconds...");
+        killProcess(self.appium, KILL_SIGNAL, function () {
+            killProcess(self.iosProxy, KILL_SIGNAL, function () {
+                setTimeout(function () {
+                    callback(e.stack);
+                }, 2000);
+            });
+        });
+    }
+
+    process.on("uncaughtException", function(err) {
+        exitGracefully(err);
+    });
+
+    logger.normal("paramedic-appium: Running tests from:");
+    self.options.testPaths.forEach(function (testPath) {
+        logger.normal("paramedic-appium: " + testPath);
+    });
+
+    jasmine.loadConfig({
+        spec_dir: "",
+        spec_files: self.options.testPaths
+    });
+
+    var medicReporter = new MedicReporter(function (results) {
+        self.cleanUp(callback);
+    });
+    // don't use default reporter, it exits the process before
+    // we would get the chance to kill appium server
```

Comment:
Could you please investigate if it is possible to modify (or override) standard Jasmine reporter instead of re-implementing the whole its functionality in 'MedicReporter.js'